src/index.js

```diff
@@ -225,11 +228,17 @@ export default class ReactSearchBox extends Component {
      * The 'value' state is updated with the clicked record's value.
      */
 
-    const { value } = record
+    let {value} = record;
+
+    if(this.props.clearOnSelect) {
+      // clear the value rather than input it into the box
+      value = "";
+    }
+
     const { onSelect, onChange } = this.props
 
     this.setState({
-      value,
+      value: value,
```

Comment:
This change doesn't look necessary.

```suggestion
      value,
```